### PR TITLE
Backport fixes from musl's libc

### PR DESCRIPTION
**Pull request by MaxGraey**
Patch references [1](https://git.musl-libc.org/cgit/musl/commit/?id=964104f9f0e056cf58d9defa0b716d7756f040f6) and [2](https://git.musl-libc.org/cgit/musl/commit/?id=c4c38e6364323b6d83ba3428464e19987b981d7a)

### CODE REVIEW

**Comment by dcodeIO** [comment]
I am not a fan of the formatting changes, as it makes PRs a little hard to grasp. I am OK with leaving it this way, though, but can you perhaps at a line comment (here on GH in the PR) with a bit of background where the code changes are and what they do?

File: std/assembly/math.ts [hunk 1]
```diff
@@ -2001,18 +2025,22 @@ export namespace NativeMathf {
...
-    if (a < 0x3F800000 + (1 << 23)) {
+    if (a < 0x3F800000 + (1 << 23)) { // |x| < 2, invalid if x < 1
       let xm1 = x - 1;
       return log1p(xm1 + builtin_sqrt(xm1 * (xm1 + 2)));
     }
-    if (a < 0x3F800000 + (12 << 23)) return log(2 * x - 1 / (x + builtin_sqrt<f32>(x * x - 1)));
+    if (u < 0x3F800000 + (12 << 23)) { // 2 <= x < 0x1p12
+      return log(2 * x - 1 / (x + builtin_sqrt<f32>(x * x - 1)));
+    }
+    // x >= 0x1p12 or x <= -2 or NaN
```
Comment (MaxGraey) [inline std/assembly/math.ts:2035]: It's main change (1)

File: std/assembly/math.ts [hunk 2]
```diff
@@ -2316,20 +2349,20 @@ export namespace NativeMathf {
...
+      Ox1p127f = reinterpret<f32>(0x7F000000); //  0x1p+127f
+
     var u = reinterpret<u32>(x);
     var hx = u & 0x7FFFFFFF;
     var sign_ = <i32>(u >> 31);
     if (hx >= 0x4195B844) {
       if (hx > 0x7F800000) return x;
       if (sign_) return -1;
-      if (x > o_threshold) {
+      if (hx > 0x42B17217) { // x > log(FLT_MAX)
```
Comment (MaxGraey) [inline std/assembly/math.ts:2365]: It's main change (2)

**Review by dcodeIO — APPROVED** [review]
Looks good, thanks! :)

**Pull request merged**